web/packages/teleterm/src/ui/DocumentTerminal/useDocumentTerminal.ts

```diff
@@ -389,6 +391,40 @@ function createCmd(
     };
   }
 
+  if (doc.kind === 'doc.gateway_kube') {
+    const gateway = clustersService.findGatewayByConnectionParams(
+      doc.targetUri,
+      ''
+    );
+    if (!gateway) {
+      throw new Error(`No gateway found for ${doc.targetUri}`);
+    }
+
+    const env = tshdGateway.getCliCommandEnv(gateway.gatewayCliCommand);
+
+    if ('KUBECONFIG' in env === false) {
+      // This shouldn't happen as 'KUBECONFIG' is the sole purpose of the CLI
+      // command for a kube gateway.
+      throw new Error(
+        `No KUBECONFIG provided for gateway ${gateway.targetUri}`
+      );
+    }
+    const helpMsg =
+      "Started local proxy for Kubernetes cluster '" +
+      gateway.targetName +
+      "' and environment variable 'KUBECONFIG' is updated to the kubeconfig provided by the local proxy.\r\n\r\n" +
+      "Try 'kubectl version' to test the connection.\r\n\r\n";
+
+    return {
+      ...doc,
```

Comment:
Spreading on the `...doc` that we do for regular `pty.shell` below is a bad practice because it's not clear what fields from `doc` are needed in `pty.shell`.

Fortunately, here we can just remove this line as AFAIK we directly pass all required fields anyway.